tap_googleads/streams.py

```diff
@@ -87,37 +109,55 @@ def gaql(self):
         )
     ).to_dict()
 
-    # Goal of this stream is to send to children stream a dict of
-    # login-customer-id:customer-id to query for all queries downstream
-    def get_records(self, context: Optional[dict]) -> Iterable[Dict[str, Any]]:
-        """Return a generator of row-type dictionary objects.
+    def post_process(
+        self,
+        row: Record,
+        context: Context | None = None,  # noqa: ARG002
+    ) -> dict | None:
+        """As needed, append or transform raw data to match expected structure.
 
-        Each row emitted should be a dictionary of property names to their values.
+        Optional. This method gives developers an opportunity to "clean up" the results
+        prior to returning records to the downstream tap - for instance: cleaning,
+        renaming, or appending properties to the raw record result returned from the
+        API.
+
+        Developers may also return `None` from this method to filter out
+        invalid or not-applicable records from the stream.
 
         Args:
+            row: Individual record in the stream.
             context: Stream partition or context dictionary.
 
-        Yields:
-            One item per (possibly processed) record in the API.
+        Returns:
+            The resulting record dict, or `None` if the record should be excluded.
+
         """
-        if self.config.get("comma_separated_string_of_customer_ids", False):
-            customer_ids_list = self.config.get("comma_separated_string_of_customer_ids").replace(" ", "").split(",")
-            for i in customer_ids_list:
-                yield {"customerClient": {"id": str(i)}}
-        else:
-            for row in self.request_records(context):
-                row = self.post_process(row, context)
-                # Don't search Manager accounts as we can't query them for everything
-                if row["customerClient"]["manager"] == True:
-                    continue
-                yield row
-
-    def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
+        customer = row["customerClient"]
+
+        if (
+            self.config.get("customer_ids")
+            and customer["id"] not in self.config["customer_ids"]
+        ):
+            return None
```

Comment:
No because this is a different level customer account I think? But hard to say how it looks in the situation where you dont sync with a manager customer account. In my case `AccessibleCustomers` only returns me my manager customer account id and then `CustomerHierarchyStream` returns all customer accounts that I have access to.

Reply:
> In my case `AccessibleCustomers` only returns me my manager customer account id and then `CustomerHierarchyStream` returns all customer accounts that I have access to.

Yeah, so in our case `AccessibleCustomers` returns both a manager and operating customer account and then `CustomerHierarchyStream` returns the operating customer account for both contexts, meaning that all reports streams are synced twice...

Reply:
Would this mean that we could just skip `AccessibleCustomers` and just use `CustomerHierarchyStream`?

Reply:
I think `AccessibleCustomers` is required to resolve customer IDs for the authenticated principal. That's why `CustomerHierarchyStream` is set up as a child stream.

Reply:
Then I think we should leave the rest of the implementation as is! I just get the manager customer account from `AccessibleCustomers` so I need the filter to happen in `CustomerHierarchyStream`

Reply:
What about https://github.com/Matatika/tap-googleads/pull/45#discussion_r1766998874?